File: test_ai.py
import sys
sys.path.insert(0, '/home/l2euser/TradeByBater')

from ai_assistant import AIOrchestrator, KnowledgeBaseProvider, ProfessionalEnglishTrainer

test_cases = [
    {"query": "hi", "expected": "Greeting"},
    {"query": "how do i list something", "expected": "Listing guide"},
    {"query": "is trading safe", "expected": "Safety info"},
    {"query": "which states are available", "expected": "Location info"},
    {"query": "is it free", "expected": "Cost info"},
    {"query": "what about scams", "expected": "Safety info"},
    {"query": "what categories exist", "expected": "Category info"},
    {"query": "getting started", "expected": "Onboarding"},
]

def run_tests():
    print("🤖 Testing TradeByBater AI System...\n")
    
    orchestrator = AIOrchestrator()
    
    for test in test_cases:
        print(f"📝 Query: \"{test['query']}\"")
        print(f"✅ Expected: {test['expected']}")
        
        response = orchestrator.process_query(test["query"])
        display = response.text[:100] + "..." if len(response.text) > 100 else response.text
        print(f"📄 Response: {display}")
        print(f"🔧 Provider: {response.provider} | Confidence: {response.confidence}\n")
    
    print("✅ All tests completed!")

if __name__ == "__main__":
    run_tests()

File: test-ai.js
async function runTests() {
  const { AIOrchestrator } = require('./lib/ai-providers');
  const { ProfessionalEnglishTrainer } = require('./lib/trainer');

  console.log("Running AI System Tests...\n");

  const orchestrator = new AIOrchestrator();
  const trainer = new ProfessionalEnglishTrainer();

  const testQueries = [
    { query: "Hi", expected: "greeting" },
    { query: "How do I list something?", expected: "listing" },
    { query: "Is it safe to trade?", expected: "safety" },
    { query: "Which states are available?", expected: "location" },
    { query: "What does it cost?", expected: "pricing" }
  ];

  for (const { query, expected } of testQueries) {
    console.log(`Q: ${query}`);
    try {
      const response = await orchestrator.processQuery(query);
      console.log(`A: ${response.text.substring(0, 100)}...`);
      console.log(`Provider: ${response.provider}, Confidence: ${response.confidence}\n`);
    } catch (error) {
      console.log(`Error: ${error.message}\n`);
    }
  }

  console.log("Tests completed. Professional English trainer demonstration:");
  const enhanced = await trainer.enhanceResponse(
    "You can list items easily.",
    { formal: true }
  );
  console.log(`Enhanced: ${enhanced}`);
}

runTests().catch(console.error);